Detectors/TRD/reconstruction/src/EventRecord.cxx

```diff
@@ -44,192 +40,58 @@
 namespace o2::trd
 {
 
-//Digit information
-std::vector<Digit>& EventRecord::getDigits() { return mDigits; }
-void EventRecord::addDigits(Digit& digit) { mDigits.push_back(digit); }
-void EventRecord::addDigits(std::vector<Digit>::iterator& start, std::vector<Digit>::iterator& end) { mDigits.insert(std::end(mDigits), start, end); }
-
-//tracklet information
-std::vector<Tracklet64>& EventRecord::getTracklets() { return mTracklets; }
-void EventRecord::addTracklet(Tracklet64& tracklet) { mTracklets.push_back(tracklet); }
-void EventRecord::addTracklets(std::vector<Tracklet64>::iterator& start, std::vector<Tracklet64>::iterator& end)
-{
-  mTracklets.insert(std::end(mTracklets), start, end);
-}
-
-void EventRecord::addTracklets(std::vector<Tracklet64>& tracklets)
-{
-  for (auto tracklet : tracklets) {
-    mTracklets.push_back(tracklet);
-  }
-}
-
-void EventRecord::sortByHCID()
+void EventRecord::sortTrackletsByHCID()
```

Comment:
The TRD tracking expects the tracklets ordered by detector ID. And it would be good to have the correct order in the output of the raw reader, because then we can move the tracklets around and don't need to duplicate them downstream. This we would need to do if we decided e.g. to do the sorting in the tracklet transformer. So indeed for me its fine to sort by detector and not HCID. In the header I made a comment that we might want to think about the sorting later. For this PR I was just doing what was already there

Reply:
I dont remember the reason for hcid sorting, but it was probably mid last year. I thought it was to make it the same as the output of the simulation or someone required the digits/tracklets to be in that order, its neither here nor there, the sort could be changed to anything, or moved out and done later.